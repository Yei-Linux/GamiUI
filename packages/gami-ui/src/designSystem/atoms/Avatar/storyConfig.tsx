import React from 'react'
import {
  getInheritGlobalStylesStories,
  getStoryConfigStructure,
} from 'core/helpers/storybook.helper'
import Avatar from '.'
import Icon from '../Icon'
import { avatarProfilesImages, options, withTextExamples } from './constants'
import { IStoryMainConfig } from 'core/domain/interfaces/IStorybook'

const mainConfig: IStoryMainConfig = {
  title: 'Atoms/Avatar',
  component: Avatar,
  args: {},
  argTypes: {
    src: {
      control: 'text',
      description: 'Image source',
      table: {
        type: { summary: 'string' },
        defaultValue: Avatar.defaultProps?.src,
      },
    },
    text: {
      control: 'text',
      description: 'Display text when image is missing',
      table: {
        type: { summary: 'string' },
        defaultValue: Avatar.defaultProps?.text,
      },
    },
    alt: {
      control: 'text',
      description: 'Display altText in avatar',
      table: {
        type: { summary: 'string' },
        defaultValue: Avatar.defaultProps?.alt,
      },
    },
    background: {
      control: 'color',
      description: 'Change avatar background',
      table: {
        type: { summary: 'string' },
        defaultValue: Avatar.defaultProps?.background,
      },
    },
    textColor: {
      control: 'color',
      description: 'Change avatar text color',
      table: {
        type: { summary: 'string' },
        defaultValue: Avatar.defaultProps?.textColor,
      },
    },
    borderColor: {
      control: 'color',
      description: 'Change avatar border color',
      table: {
        type: { summary: 'string' },
        defaultValue: Avatar.defaultProps?.borderColor,
      },
    },
    zoomMode: {
      control: { type: 'select', options: options.zoomMode },
      description: 'Display avatar zoom',
      table: {
        type: { summary: 'string' },
        defaultValue: Avatar.defaultProps?.zoomMode,
      },
    },
    textMode: {
      control: { type: 'select', options: options.textMode },
      description: 'Display avatar text mode',
      table: {
        type: { summary: 'string' },
        defaultValue: Avatar.defaultProps?.textMode,
      },
    },
  },
}

const storiesInheritGlobalStyles = getInheritGlobalStylesStories({
  border: {
    args: {
      background: '#F76E11',
      src: avatarProfilesImages.one,
      zoomMode: 'inside',
    },
  },
  shadow: {
    args: {
      src: avatarProfilesImages.one,
      zoomMode: 'inside',
      border: 'lg',
    },
  },
  width: {
    args: {
      src: avatarProfilesImages.one,
      height: 'auto',
      zoomMode: 'inside',
      border: 'lg',
    },
  },
  height: {
    args: {
      src: avatarProfilesImages.one,
      width: 'sm',
      zoomMode: 'inside',
      border: 'lg',
    },
  },
  size: {
    args: {
      src: avatarProfilesImages.one,
      zoomMode: 'inside',
      border: 'lg',
    },
  },
  fontWeight: null,
  textAlign: null,
  margin: {
    args: {
      background: '#F76E11',
      src: avatarProfilesImages.one,
      zoomMode: 'inside',
      border: 'lg',
    },
    examples: ['0px', '2px', '4px', '6px', '8px', '10px'],
  },
  padding: {
    args: {
      background: '#F76E11',
      src: avatarProfilesImages.one,
      zoomMode: 'inside',
      border: 'lg',
    },
    examples: ['0px', '2px', '4px', '6px', '8px', '10px'],
  },
})

const storiesComponent = [
  {
    storyName: 'WithGroup',
    parent: {
      args: {
        count: 12,
      },
      variants: [
        {
          label: 'WithImage',
          value: Object.entries(avatarProfilesImages).map(
            ([, value], index) => (
              <Avatar zoomMode="outside" src={value} key={index} />
            )
          ),
        },
        {
          label: 'WithText',
          value: withTextExamples.map(({ value, customProps }, index) => (
            <Avatar
              zoomMode="outside"
              text={value}
              key={index}
              {...customProps}
            />
          )),
        },
      ],
    },
  },
  {
    storyName: 'WithImage',
    self: {
      args: {
        zoomMode: 'inside',
      },
      variants: {
        examples: Object.entries(avatarProfilesImages).map(
          ([, value], index) => ({
            label: `Image ${index + 1}`,
            value,
          })
        ),
        field: 'src',
      },
    },
  },
  {
    storyName: 'WithText',
    self: {
      args: {},
      variants: {
        examples: withTextExamples,
        field: 'text',
      },
    },
  },
  {
    storyName: 'WithIcon',
    self: {
      args: {
        background: 'white',
        shadow: 'md',
      },
      variants: {
        examples: [
          {
            label: 'Icon 1',
            value: (
              <Icon fill="none" size="25px" name="facebook__transparent" />
            ),
          },
          {
            label: 'Icon 2',
            value: (
              <Icon fill="none" size="25px" name="facebook__transparent" />
            ),
          },
          {
            label: 'Icon 3',
            value: (
              <Icon fill="none" size="25px" name="facebook__transparent" />
            ),
          },
        ],
        field: 'icon',
      },
    },
  },
  {
    storyName: 'WithAlt',
    self: {
      args: {
        background: '#F76E11',
        src: avatarProfilesImages.one,
        zoomMode: 'inside',
        shadow: 'md',
      },
      variants: {
        examples: [
          {
            label: 'Alt 1',
            value: 'Example Alt Avatar 1',
          },
          {
            label: 'Alt 2',
            value: 'Example Alt Avatar 2',
          },
          {
            label: 'Alt 3',
            value: 'Example Alt Avatar 3',
          },
        ],
        field: 'alt',
      },
    },
  },
  {
    storyName: 'WithSetsAndBreakpoints',
    self: {
      args: {
        breakpoints: [
          { mediaMaxWidth: '320px', size: '280px' },
          { mediaMaxWidth: '480px', size: '440px' },
          { size: '800px' },
        ],
        background: '#F76E11',
        src: avatarProfilesImages.one,
        zoomMode: 'inside',
        shadow: 'md',
      },
      variants: {
        examples: [
          {
            label: 'For Sets device',
            value: [
              { setSrc: avatarProfilesImages.one, setWidth: '320w' },
              { setSrc: avatarProfilesImages.one, setWidth: '480w' },
              { setSrc: avatarProfilesImages.one, setWidth: '800w' },
            ],
          },
        ],
        field: 'sets',
      },
    },
  },
  {
    storyName: 'WithZoomMode',
    self: {
      args: {
        background: '#F76E11',
        src: avatarProfilesImages.one,
      },
      variants: {
        examples: options.zoomMode.map((mode) => ({
          label: mode,
          value: mode,
        })),
        field: 'zoomMode',
      },
    },
  },
  {
    storyName: 'WithTextMode',
    self: {
      args: {
        text: 'Jesus Alvan',
      },
      variants: {
        examples: options.textMode.map((mode) => ({
          label: `${mode}: (Jesus Alvan)`,
          value: mode,
        })),
        field: 'textMode',
      },
    },
  },
]

const storyConfig = getStoryConfigStructure({
  mainConfig,
  storiesInheritGlobalStyles,
  storiesComponent,
  component: Avatar,
  parentComponent: Avatar.Group,
})

export { storyConfig }
