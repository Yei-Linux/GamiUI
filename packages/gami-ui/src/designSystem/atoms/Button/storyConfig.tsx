import React from 'react'
import { IStoryMainConfig } from 'core/domain/interfaces/IStorybook'
import {
  getInheritGlobalStylesStories,
  getStoryConfigStructure,
} from 'core/helpers/storybook.helper'
import Button, { options } from '.'
import Icon from '../Icon'

const mainConfig: IStoryMainConfig = {
  title: 'Atoms/Button',
  component: Button,
  args: {},
  argTypes: {},
}

const storiesInheritGlobalStyles = getInheritGlobalStylesStories({
  rounded: {
    args: {
      children: 'Gami Button',
      fontWeight: 'semibold',
    },
  },
  shadow: {
    args: {
      children: 'Gami Button',
      fontWeight: 'semibold',
    },
  },
  width: {
    args: {
      children: 'GamiUI',
      fontWeight: 'semibold',
    },
  },
  height: {
    args: {
      children: 'GamiUI',
      fontWeight: 'semibold',
    },
  },
  size: {
    args: {
      children: 'GamiUI',
      fontWeight: 'semibold',
    },
  },
  fontWeight: {
    args: {
      children: 'GamiUI',
    },
  },
  textAlign: {
    args: {
      width: 'lg',
      children: 'GamiUI',
    },
  },
  margin: {
    args: {
      children: 'Gami Button',
    },
    examples: ['0px', '2px', '4px', '6px', '8px', '10px'],
  },
  padding: {
    args: {
      children: 'Gami Button',
    },
    examples: ['0px', '2px', '4px', '6px', '8px', '10px'],
  },
})

const storiesComponent = [
  {
    storyName: 'WithDisable',
    self: {
      args: {
        children: 'Gami Button',
        shadow: 'flat',
      },
      variants: {
        examples: [
          {
            label: 'Disabled',
            value: true,
          },
          {
            label: 'Enable',
            value: false,
          },
        ],
        field: 'disable',
      },
    },
  },
  {
    storyName: 'WithVariants',
    self: {
      args: {
        children: 'Gami Button',
        fontWeight: 'semibold',
      },
      variants: {
        examples: options.type.map((variant) => ({
          label: variant,
          value: variant,
          customProps: {
            shadow: variant,
          },
        })),
        field: 'variant',
      },
    },
  },
  {
    storyName: 'WithBordered',
    self: {
      args: {
        children: 'Gami Button',
        bordered: true,
        fontWeight: 'semibold',
      },
      variants: {
        examples: options.type.map((variant) => ({
          label: variant,
          value: variant,
        })),
        field: 'variant',
      },
    },
  },
  {
    storyName: 'WithGhost',
    self: {
      args: {
        children: 'Gami Button',
        ghost: true,
        fontWeight: 'semibold',
      },
      variants: {
        examples: options.type.map((variant) => ({
          label: variant,
          value: variant,
        })),
        field: 'variant',
      },
    },
  },
  {
    storyName: 'WithLight',
    self: {
      args: {
        children: 'Gami Button',
        shadow: 'none',
        light: true,
        fontWeight: 'semibold',
      },
      variants: {
        examples: options.type.map((variant) => ({
          label: variant,
          value: variant,
        })),
        field: 'variant',
      },
    },
  },
  {
    storyName: 'WithFlat',
    self: {
      args: {
        children: 'Gami Button',
        shadow: 'none',
        flat: true,
        fontWeight: 'semibold',
      },
      variants: {
        examples: options.type.map((variant) => ({
          label: variant,
          value: variant,
        })),
        field: 'variant',
      },
    },
  },
  {
    storyName: 'WithPreffix',
    self: {
      args: {
        children: 'Gami Button',
        variant: 'info',
        shadow: 'info',
        flat: true,
        fontWeight: 'semibold',
      },
      variants: {
        examples: [
          {
            label: 'Icon 1',
            value: <Icon fill="none" size="25px" name="edit" />,
          },
          {
            label: 'Icon 2',
            value: <Icon fill="none" size="25px" name="edit" />,
          },
          {
            label: 'Icon 3',
            value: <Icon fill="none" size="25px" name="edit" />,
          },
        ],
        field: 'preffix',
      },
    },
  },
  {
    storyName: 'WithHtmlButtonTypes',
    self: {
      args: {
        children: 'Gami Button',
        fontWeight: 'semibold',
      },
      variants: {
        examples: options.typeHtml.map((typeHtml) => ({
          label: typeHtml,
          value: typeHtml,
        })),
        field: 'type',
      },
    },
  },
]

const storyConfig = getStoryConfigStructure({
  mainConfig,
  storiesInheritGlobalStyles,
  storiesComponent,
  component: Button,
})

export { storyConfig }
