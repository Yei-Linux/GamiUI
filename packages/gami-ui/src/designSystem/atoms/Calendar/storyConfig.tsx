import React from 'react'
import {
  IStoryConfig,
  IStoryMainConfig,
} from 'core/domain/interfaces/IStorybook'
import {
  getInheritGlobalStylesStories,
  getStoryConfigStructure,
} from 'core/helpers/storybook.helper'
import Calendar from '.'
import { CalendarWithInteractionForStorybook } from './mock'
import { dateHelper } from 'core/helpers/date.helper'

const storyArgTypes = {
  className: {
    control: 'text',
    description: 'Display className prefix',
    table: {
      type: { summary: 'string' },
    },
  },
  currentDate: {
    control: 'object',
    description: 'Get the current date and manage it in the calendar',
    table: {
      type: { summary: 'date' },
    },
  },
  daySelected: {
    control: 'number',
    description: 'Get the current timestamp and select the day in the calendar',
    table: {
      type: { summary: 'number' },
    },
  },
}

const docArgTypes = {
  handleSelectCurrentDate: {
    control: 'object',
    description:
      'Function to handler when is going either to next or prev month',
    table: { type: { summary: 'object' } },
  },
  handleSelectDay: {
    control: 'object',
    description: 'Function to handler when is selecting a day into calendar',
    table: { type: { summary: 'object' } },
  },
}

const argTypes = { ...storyArgTypes, ...docArgTypes }

const mainConfig: IStoryMainConfig = {
  title: 'Atoms/Calendar ðŸŸ¢',
  component: Calendar,
  args: {},
  argTypes: storyArgTypes,
}

const storiesInheritGlobalStyles = getInheritGlobalStylesStories({
  rounded: null,
  shadow: null,
  margin: null,
  padding: null,
  width: null,
  height: null,
  size: null,
  fontWeight: null,
  textAlign: null,
})

const storiesComponent = [] as IStoryConfig[]

const dateManager = dateHelper({ date: new Date(2022, 10, 12) })
const customStories = [
  {
    storyBuilder: (args: any) => (
      <CalendarWithInteractionForStorybook {...args} />
    ),
    storyName: 'With Interaction ðŸ™‚',
    args: { defaultValue: new Date(2022, 10, 12) },
  },
  {
    storyBuilder: (args: any) => <Calendar {...args} />,
    storyName: 'With Current Date ðŸ™‚',
    args: {
      currentDate: dateManager.date,
      daySelected: dateManager.getTimeStampByDate,
    },
  },
]

const storyConfig = getStoryConfigStructure({
  mainConfig,
  storiesInheritGlobalStyles,
  storiesComponent,
  customStories,
  component: Calendar,
})

export { storyConfig, argTypes }
