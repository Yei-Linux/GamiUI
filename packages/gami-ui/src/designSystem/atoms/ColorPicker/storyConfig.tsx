import React from 'react'
import {
  IStoryConfig,
  IStoryMainConfig,
} from 'core/domain/interfaces/IStorybook'
import {
  getInheritGlobalStylesStories,
  getStoryConfigStructure,
} from 'core/helpers/storybook.helper'
import ColorPicker from '.'
import { WithColorPickerInteraction } from './mock'

const storyArgTypes = {
  value: {
    control: 'text',
    description: 'Value to display it on input picker',
    table: {
      type: { summary: 'string' },
    },
  },
}

const docArgTypes = {
  onChangeFormItem: {
    control: 'object',
    description: 'Function to handler when is changing color',
    table: { type: { summary: 'object' } },
  },
}

const argTypes = { ...storyArgTypes, ...docArgTypes }

const mainConfig: IStoryMainConfig = {
  title: 'Atoms/ColorPicker ðŸŸ¢',
  component: ColorPicker,
  args: {},
  argTypes: storyArgTypes,
}

const storiesInheritGlobalStyles = getInheritGlobalStylesStories({
  rounded: null,
  shadow: null,
  margin: null,
  padding: null,
  width: null,
  height: null,
  size: null,
  fontWeight: null,
  textAlign: null,
})

const storiesComponent = [] as IStoryConfig[]

const customStories = [
  {
    storyBuilder: (args: any) => <WithColorPickerInteraction {...args} />,
    storyName: 'With Interaction ðŸ™‚',
    args: {},
  },
]

const storyConfig = getStoryConfigStructure({
  mainConfig,
  storiesInheritGlobalStyles,
  storiesComponent,
  customStories,
  component: ColorPicker,
})

export { storyConfig, argTypes }
