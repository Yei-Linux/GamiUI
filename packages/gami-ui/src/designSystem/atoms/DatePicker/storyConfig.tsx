import React from 'react'
import {
  IStoryConfig,
  IStoryMainConfig,
} from 'core/domain/interfaces/IStorybook'
import {
  getInheritGlobalStylesStories,
  getStoryConfigStructure,
} from 'core/helpers/storybook.helper'
import { WithBasic, WithTooltipAbove, WithDefault } from './mock'
import DatePicker from './DatePicker'

const storyArgTypes = {
  className: {
    control: 'text',
    description: 'Classname preffix to date picker',
    table: {
      type: { summary: 'string' },
    },
  },
  value: {
    control: 'text',
    description: 'Value to display it on date picker',
    table: {
      type: { summary: 'string' },
    },
  },
  formatter: {
    control: 'text',
    description: 'formatter on date picker',
    table: {
      type: { summary: 'string' },
    },
  },
}

const docArgTypes = {
  onChangeFormItem: {
    control: 'object',
    description: 'Function to handler when is changing date',
    table: { type: { summary: 'object' } },
  },
}

const argTypes = { ...storyArgTypes, ...docArgTypes }

const mainConfig: IStoryMainConfig = {
  title: 'Atoms/DatePicker ðŸŸ¢',
  component: DatePicker,
  args: {},
  argTypes: storyArgTypes,
}

const storiesInheritGlobalStyles = getInheritGlobalStylesStories({
  rounded: null,
  shadow: null,
  margin: null,
  padding: null,
  width: null,
  height: null,
  size: null,
  fontWeight: null,
  textAlign: null,
})

const storiesComponent = [] as IStoryConfig[]

const customStories = [
  {
    storyBuilder: (args: any) => <WithBasic {...args} />,
    storyName: 'With Basic Interaction ðŸ™‚',
    args: {},
  },
  {
    storyBuilder: (args: any) => <WithTooltipAbove {...args} />,
    storyName: 'With TooltipAbove ðŸ™‚',
    args: {},
  },
  {
    storyBuilder: (args: any) => <WithDefault {...args} />,
    storyName: 'WithDefault ðŸ™‚',
    args: {},
  },
]

const storyConfig = getStoryConfigStructure({
  mainConfig,
  storiesInheritGlobalStyles,
  storiesComponent,
  customStories,
  component: DatePicker,
})

export { storyConfig, argTypes }
