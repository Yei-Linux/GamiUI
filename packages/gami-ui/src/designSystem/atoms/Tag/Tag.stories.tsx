import { Meta } from '@storybook/react'

import { storyConfig } from './storyConfig'

const { mainConfig, stories } = storyConfig

export default mainConfig as Meta

const [one, two, three, four, five, six, seven, eight, nine] = stories

export { one, two, three, four, five, six, seven, eight, nine }
