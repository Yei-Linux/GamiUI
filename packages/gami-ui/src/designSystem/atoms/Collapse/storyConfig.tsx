import React from 'react'
import { IStoryMainConfig } from 'core/domain/interfaces/IStorybook'
import {
  getInheritGlobalStylesStories,
  getStoryConfigStructure,
  madegenericPropsControl,
} from 'core/helpers/storybook.helper'
import Collapse from '.'
import RichText from '../RichText'
import Title from '../Title'
import Avatar from '../Avatar'
import { options } from '../Button'
import { TitleCollapses } from './constants'

const genericArgTypes = madegenericPropsControl([
  'size',
  'fontWeight',
  'textAlign',
  'height',
  'width',
])

const storyArgTypes = {
  variant: {
    control: { type: 'select', options: options.type },
    description: 'Display theme button variants',
    table: {
      type: { summary: 'string' },
      defaultValue: Collapse.defaultProps?.variant,
    },
  },
  bordered: {
    control: 'boolean',
    description: 'Set bordered or not',
    table: {
      type: { summary: 'boolean' },
      defaultValue: Collapse.defaultProps?.bordered,
    },
  },
  expanded: {
    control: 'boolean',
    description: 'Set default visible',
    table: {
      type: { summary: 'boolean' },
      defaultValue: Collapse.defaultProps?.expanded,
    },
  },
  ...genericArgTypes,
}

const docArgTypes = {
  title: {
    control: 'object',
    description: 'Display title collapse',
    table: {
      type: { summary: 'React.ReactNode' },
    },
  },
  subtitle: {
    control: 'object',
    description: 'Display subtitle collapse',
    table: {
      type: { summary: 'React.ReactNode' },
    },
  },
  contentLeft: {
    control: 'object',
    description: 'Display content Left of Collapse Header',
    table: {
      type: { summary: 'React.ReactNode' },
    },
  },
  icon: {
    control: 'object',
    description: 'Display icon of Collapse Header',
    table: {
      type: { summary: 'React.ReactNode' },
    },
  },
  children: {
    control: 'object',
    description: 'Display content Of Collapse',
    table: {
      type: { summary: 'React.ReactNode' },
    },
  },
}

const argTypes = { ...storyArgTypes, ...docArgTypes }

const mainConfig: IStoryMainConfig = {
  title: 'Atoms/Collapse',
  component: Collapse,
  args: {},
  argTypes: storyArgTypes,
}

const textContentMock = () => (
  <RichText text="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat." />
)
const storiesInheritGlobalStyles = getInheritGlobalStylesStories({
  rounded: {
    args: {
      style: { maxWidth: '300px' },
      title: <Title level="h3">Title</Title>,
      subtitle: 'Subtitle',
      children: textContentMock(),
      shadow: 'xs',
    },
  },
  shadow: {
    args: {
      style: { maxWidth: '300px' },
      title: <Title level="h3">Title</Title>,
      subtitle: 'Subtitle',
      children: textContentMock(),
    },
  },
  margin: {
    args: {
      style: { maxWidth: '300px' },
      title: <Title level="h3">Title</Title>,
      subtitle: 'Subtitle',
      children: textContentMock(),
      shadow: 'xs',
    },
    examples: ['0px', '2px', '4px', '6px', '8px', '10px'],
  },
  padding: {
    args: {
      style: { maxWidth: '300px' },
      title: <Title level="h3">Title</Title>,
      subtitle: 'Subtitle',
      children: textContentMock(),
      shadow: 'xs',
    },
    examples: ['0px', '2px', '4px', '6px', '8px', '10px'],
  },
  width: null,
  height: null,
  size: null,
  fontWeight: null,
  textAlign: null,
})

const storiesComponent = [
  {
    storyName: 'WithGroupAndDivider',
    parent: {
      args: { divider: true },
      variants: [
        {
          label: 'WithDivider',
          value: TitleCollapses.map((Title) => ({
            props: { title: Title, children: textContentMock() },
            Component: Collapse,
          })),
        },
      ],
    },
  },
  {
    storyName: 'WithGroupAndNotDivider',
    parent: {
      args: { divider: false },
      variants: [
        {
          label: 'WithNotDivider',
          value: TitleCollapses.map((Title) => ({
            props: { title: Title, children: textContentMock() },
            Component: Collapse,
          })),
        },
      ],
    },
  },
  {
    storyName: 'WithGroupAndAccordion',
    parent: {
      args: { accordion: true },
      variants: [
        {
          label: 'WithAccordion',
          value: TitleCollapses.map((Title) => ({
            props: { title: Title, children: textContentMock() },
            Component: Collapse,
          })),
        },
      ],
    },
  },
  {
    storyName: 'WithTitle',
    self: {
      args: {
        children: textContentMock(),
      },
      variants: {
        examples: [
          {
            label: 'Title A',
            value: <Title level="h3">Title A</Title>,
          },
          {
            label: 'Title B',
            value: <Title level="h3">Title B</Title>,
          },
          {
            label: 'Title C',
            value: <Title level="h3">Title C</Title>,
          },
        ],
        field: 'title',
      },
    },
  },
  {
    storyName: 'WithSubTitle',
    self: {
      args: {
        title: <Title level="h3">Title</Title>,
        children: textContentMock(),
      },
      variants: {
        examples: [
          {
            label: 'Subtitle A',
            value: 'Subtitle A',
          },
          {
            label: 'Subtitle B',
            value: 'Subtitle B',
          },
          {
            label: 'Subtitle C',
            value: 'Subtitle C',
          },
        ],
        field: 'subtitle',
      },
    },
  },
  {
    storyName: 'WithExpanded',
    self: {
      args: {
        title: <Title level="h3">Title</Title>,
        subtitle: 'Subtitle',
        children: textContentMock(),
      },
      variants: {
        examples: [
          {
            label: 'Initial Expanded',
            value: true,
          },
          {
            label: 'Not initial expanded',
            value: false,
          },
        ],
        field: 'expanded',
      },
    },
  },
  {
    storyName: 'WithVariants',
    self: {
      args: {
        title: <Title level="h2">Title</Title>,
        subtitle: 'Subtitle',
        rounded: 'md',
        children: textContentMock(),
      },
      variants: {
        examples: options.type.map((variant) => ({
          label: variant,
          value: variant,
          customProps: {
            shadow: variant,
          },
        })),
        field: 'variant',
      },
    },
  },
  {
    storyName: 'WithBordered',
    self: {
      args: {
        title: <Title level="h2">Title</Title>,
        subtitle: 'Subtitle',
        rounded: 'md',
        bordered: true,
        children: textContentMock(),
      },
      variants: {
        examples: options.type.map((variant) => ({
          label: variant,
          value: variant,
        })),
        field: 'variant',
      },
    },
  },
  {
    storyName: 'WithContentLeft',
    self: {
      args: {
        title: <Title level="h3">Title</Title>,
        subtitle: 'Subtitle',
        children: textContentMock(),
      },
      variants: {
        examples: [
          {
            label: 'Exm A',
            value: (
              <Avatar
                rounded="lg"
                shadow="flat"
                src="https://i.imgur.com/vyW3w3r.png"
                zoomMode="inside"
              />
            ),
          },
          {
            label: 'Exm B',
            value: (
              <Avatar
                rounded="lg"
                shadow="flat"
                src="https://i.imgur.com/lkIgoZP.png"
                zoomMode="inside"
              />
            ),
          },
          {
            label: 'Exm C',
            value: (
              <Avatar
                rounded="lg"
                shadow="flat"
                src="https://i.imgur.com/IXPZxIr.png"
                zoomMode="inside"
              />
            ),
          },
        ],
        field: 'contentLeft',
      },
    },
  },
]

const storyConfig = getStoryConfigStructure({
  mainConfig,
  storiesInheritGlobalStyles,
  storiesComponent,
  component: Collapse,
  parentComponent: Collapse.Group,
})

export { storyConfig, argTypes }
