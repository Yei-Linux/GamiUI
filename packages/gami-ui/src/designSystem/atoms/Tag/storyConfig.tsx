import { IStoryMainConfig } from 'core/domain/interfaces/IStorybook'
import {
  getInheritGlobalStylesStories,
  getStoryConfigStructure,
} from 'core/helpers/storybook.helper'
import Tag from '.'

const storyArgTypes = {
  text: {
    control: 'text',
    description: 'Text on tag',
    table: {
      type: { summary: 'string' },
      defaultValue: Tag.defaultProps?.text,
    },
  },
  color: {
    control: 'text',
    description: 'Color on tag',
    table: {
      type: { summary: 'string' },
      defaultValue: Tag.defaultProps?.color,
    },
  },
  background: {
    control: 'text',
    description: 'Background on tag',
    table: {
      type: { summary: 'string' },
      defaultValue: Tag.defaultProps?.background,
    },
  },
  ballMarkerColor: {
    control: 'text',
    description: 'Ball Marker Color',
    table: {
      type: { summary: 'string' },
      defaultValue: Tag.defaultProps?.ballMarkerColor,
    },
  },
}

const docArgTypes = {}

const argTypes = { ...storyArgTypes, ...docArgTypes }

const mainConfig: IStoryMainConfig = {
  title: 'Atoms/Tag',
  component: Tag,
  args: {},
  argTypes: storyArgTypes,
}

const storiesInheritGlobalStyles = getInheritGlobalStylesStories({
  rounded: {
    args: {
      text: 'Gami Tag',
    },
  },
  shadow: {
    args: {
      text: 'Gami Tag',
    },
  },
  width: {
    args: {
      text: 'Gami Tag',
    },
  },
  height: {
    args: {
      text: 'Gami Tag',
    },
  },
  fontWeight: {
    args: {
      text: 'Gami Tag',
    },
  },
  textAlign: {
    args: {
      text: 'Gami Tag',
    },
  },
  margin: {
    args: {
      text: 'Gami Tag',
    },
    examples: ['0px', '2px', '4px', '6px', '8px', '10px'],
  },
  padding: {
    args: {
      text: 'Gami Tag',
    },
    examples: ['0px', '2px', '4px', '6px', '8px', '10px'],
  },
  size: null,
})

const storiesComponent = [
  {
    storyName: 'WithColorsAndMarker',
    self: {
      args: {
        text: 'Gami Tag',
        ballMarkerColor: '#7f9cf5',
      },
      variants: {
        examples: [
          {
            label: 'Color A',
            value: '#9879e9',
          },
          {
            label: 'Color B',
            value: '#54BAB9',
          },
        ],
        field: 'color',
      },
    },
  },
]

const storyConfig = getStoryConfigStructure({
  mainConfig,
  storiesInheritGlobalStyles,
  storiesComponent,
  component: Tag,
})

export { storyConfig, argTypes }
