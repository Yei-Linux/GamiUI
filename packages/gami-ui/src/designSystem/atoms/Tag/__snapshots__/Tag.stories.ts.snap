// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Tag With Marker 1`] = `
<div
  className="css-g8smug-TagWrapper ecq9i40"
  color="#A4A4A4"
  fontWeight="BOLD"
>
  <span
    className="css-p5qp9n-BallMarker ecq9i41"
  />
  Online
</div>
`;

exports[`Storyshots Atoms/Tag With Only Text 1`] = `
<div
  className="css-ujihpl-TagWrapper ecq9i40"
  color="#A4A4A4"
  fontWeight="BOLD"
>
  Multiplayer
</div>
`;
